lib/pbench/test/unit/agent/test_tool_meister.py

```diff
@@ -2,15 +2,59 @@
 Tests for the Tool Data Meister modules.
 """
 import pathlib
-import tarfile
+import subprocess
 import pytest
 
 from pbench.agent.tool_meister import ToolMeister, get_logger
 
+logger = get_logger("__logger__")
+
+
+@pytest.fixture()
+def mock_tar(monkeypatch):
+    def fake_run(*args, **kwargs):
+        def f():
+            return
+
+        f.returncode = 0
+        f.stdout = b""
+        return f
+
+    monkeypatch.setattr(subprocess, "run", fake_run)
```

Comment:
Huh? When you call `subprocess.run` you get back a *function* with `returncode` and `stdout` properties? I don't understand the logic here.

I'd suggest returning a `CompletedProcess` (or at least something that quacks like a `CompletedProcess` allowing the caller to access the necessary properties.

Reply:
> Huh? When you call `subprocess.run` you get back a _function_ with `returncode` and `stdout` properties? I don't understand the logic here.

Yes the returned function has `returncode` and `stdout` attributes attached to it.

> I'd suggest returning a `CompletedProcess` (or at least something that quacks like a `CompletedProcess` allowing the caller to access the necessary properties.

True, I can return something like `subprocess.CompletedProcess(args, returncode=0, stdout=b"", stderr=None)`